API_MANIFESTO.md

```diff
@@ -0,0 +1,91 @@
+# The API Manifesto
+
+# Infinispan API dependency module
+
+The ```infinispan-api``` dependency will contain the public API. This module will be independent from the other modules. 
+Dependency with implementations will be **logical**. This will keep the modularity needed for the different use cases.
+
+Examples:
+ * Cache use case
+ 
+ * NoSQL data store use case
+ 
+ * Tasks, concurrency, counters uses cases
+
+The dependencies to thirty-party libraries must be avoided. This module should contain dependencies to JDK classes, 
+not more.
+
+# User First
+
+* Before making an API public, consider to test the usability with someone not familiar with the feature
```

Comment:
I think it's great to have the user centric perspective in the manifesto. One comment I'd make in the spirit of this section is that we should put more emphasis on use cases over features. Features are quite product specific sets of functionality, whereas users should care more about the problem they're trying to solve or whatever need they might have. Focus goes on what the user wants to achieve, not what the product does.

This comment might not be entirely relevant here. I thought I'd mention it for consideration.

Reply:
I totally agree with what you are saying here. And I'm going to add it in the document